src/include/sof/probe/probe.h

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __SOF_PROBE_PROBE_H__
+#define __SOF_PROBE_PROBE_H__
+
+#include <ipc/probe.h>
+
+/*
+ * \brief Initialize probes subsystem
+ *
+ * param[in,optional] extraction_probe_dma - DMA associated with extraction
+ *		      In case extraction_probe_dma is NULL, extraction probes
+ *		      are unavailable.
+ */
+int probe_init(struct probe_dma *extraction_probe_dma);
+
+/*
+ * \brief Deinitialize probes subsystem.
+ *
+ * Detach extraction DMA if was enabled. Return -EINVAL in case some probes
+ * are still in use.
+ */
+int probe_deinit(void);
+
+/*
+ * \brief Setup injection DMAs for probes.
+ *
+ * param[in] count - number of DMAs configured during this call
+ * param[in] probe_dma - Array of size 'count' with configuration data for DMAs
+ */
+int probe_dma_set(uint32_t count, struct probe_dma *probe_dma);
+
+/*
+ * \brief Get info about connected injection DMAs
+ *
+ * param[in,out] data - reply to write data to
+ * param[in] max_size - maximum number of bytes available in data
+ */
+int probe_dma_get(struct sof_ipc_probe_get_params *data, uint32_t max_size);
```

Comment:
what happens if the size if not large enough to provide all the data?

Reply:
In cAVS such behavior is undefined - however, due to large mailbox size - PAGE_SIZE:4096 - and small descriptor size - 3x 4bytes - the out-of-range may never occur.
Things may be different in SOF, maybe?

Reply:
What I meant is 
A) would an error be reported or
B) would a partial set of data be provided?

Reply:
For probes, error should be returned, option a).